ambry-clustermap/src/test/java/com/github/ambry/clustermap/HelixParticipantTest.java

```diff
@@ -301,21 +307,45 @@ public void testBadCases() throws IOException {
     props.setProperty("clustermap.dcs.zk.connect.strings", "");
     clusterMapConfig = new ClusterMapConfig(new VerifiableProperties(props));
     try {
-      new HelixParticipant(clusterMapConfig, helixManagerFactory, new MetricRegistry());
+      new HelixClusterAgentsFactory(clusterMapConfig, new MetricRegistry()).getClusterParticipants();
       fail("Instantiation should have failed");
     } catch (IOException e) {
       // OK
     }
   }
 
+  /**
+   * Test instantiating multiple {@link HelixParticipant}(s) in {@link HelixClusterAgentsFactory}
+   * @throws Exception
+   */
+  @Test
+  public void testMultiParticipants() throws Exception {
+    JSONArray zkInfosJson = new JSONArray();
+    // create a new zkJson which contains two zk endpoints in the same data center.
+    JSONObject zkInfoJson = new JSONObject();
+    zkInfoJson.put(ClusterMapUtils.DATACENTER_STR, "DC0");
+    zkInfoJson.put(ClusterMapUtils.DATACENTER_ID_STR, (byte) 0);
+    zkInfoJson.put(ClusterMapUtils.ZKCONNECT_STR, "localhost:2199" + ZKCONNECT_STR_DELIMITER + "localhost:2299");
+    zkInfosJson.put(zkInfoJson);
+    JSONObject jsonObject = new JSONObject().put(ClusterMapUtils.ZKINFO_STR, zkInfosJson);
+    props.setProperty("clustermap.dcs.zk.connect.strings", jsonObject.toString(2));
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(new VerifiableProperties(props));
+    List<ClusterParticipant> participants =
+        new HelixClusterAgentsFactory(clusterMapConfig, helixManagerFactory).getClusterParticipants();
+    assertEquals("Number of participants is not expected", 2, participants.size());
+    // restore previous setup
+    props.setProperty("clustermap.dcs.zk.connect.strings", zkJson.toString(2));
```

Comment:
Minor: It's generally better to let either the test constructor or @Before method do this kind of initialization in case an exception happens in test case.